backend/src/custom.d.ts

```diff
@@ -0,0 +1,11 @@
+// Adds additional params to default express.Request type
+declare namespace Express {
+  export interface Request {
+    // In unauthenticated cases, user is not appended to req.
+    // However it is troublesome to check for presence of user object,
+    // hence type is added as a required
+    user: {
```

Comment:
adds type for `req.user` globally

Reply:
There's a type in middlewares called `CallHomeRequest` for this reason. Would that be good enough?

Reply:
yea i guess so, just that we will have to explicitly give the type each time req appears. allows more granular control of the type though. lets continue with using just `CallHomeRequest`

Reply:
Agreed, perhaps we should just use `CallHomeRequest` for these route handlers.

The reason why I made this two different types is because `req.user` is only included under certain conditions, so making them distinct in the code might pay dividends. Unfortunately, I can't figure out a way to make express figure out the types via the middlewares.